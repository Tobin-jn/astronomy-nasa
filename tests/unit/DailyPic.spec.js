import { shallowMount, mount } from "@vue/test-utils";
import DailyPic from "@/views/DailyPic.vue";

describe('DailyPic', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(DailyPic)
    expect(wrapper.isVueInstance()).toBeTruthy()
  });
  it('renders correctly', () => {
    const wrapper = mount(DailyPic)
    expect(wrapper.element).toMatchSnapshot()
  });

  // it('should get the date in the correct format', () => {
  //   let today = Date
  //   Date = jest.fn().mockImplementation(() => { return '2019-01-09T17:31:27.301Z'})
  //   today.getFullYear = jest.fn().mockImplementation(() => { return '2019'})
    
  //   const wrapper = mount(DailyPic)
  //   expect(wrapper.vm.getDay()).toEqual('')
  // });
});

   