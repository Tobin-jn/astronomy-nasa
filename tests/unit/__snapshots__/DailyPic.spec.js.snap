// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyPic renders correctly 1`] = `
<div
  class="dailypic"
>
  <h1>
    Astronomy Picture of the Day
  </h1>
   
  <router-link
    to="/"
  >
    <img
      alt="Home Icon"
      class="home-btn"
      src="../assets/home-icon.png"
    />
  </router-link>
   
  <div
    class="pic-container"
  >
    <h3
      class="date"
    >
      January 9, 2019
    </h3>
     
    <img
      alt="Daily Astronomy Picture"
      class="image"
      src="https://apod.nasa.gov/apod/image/1901/sombrero_spitzer_3000.jpg"
    />
     
    <p
      class="explanation"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
  </div>
   
  <button
    class="monthly-pic-btn"
  >
    <router-link
      to="/MonthlyPics"
    >
      Pictures of the Month
    </router-link>
  </button>
</div>
`;
